src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheckTest.java

```diff
@@ -220,4 +220,22 @@ public void testAtclause() throws Exception {
         verify(checkConfig,
                 getPath("InputAtclauseOrderWithAnnotationsOutsideJavadoc.java"), expected);
     }
+
+    @Test
+    public void testNewArrayDeclaratorStructure() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(AtclauseOrderCheck.class);
+        final String tagOrder = "[@author, @version, @param, @return, @throws, @exception, @see,"
+                + " @since, @serial, @serialField, @serialData, @deprecated]";
+
+        final String[] expected = {
+            "32: " + getCheckMessage(MSG_KEY, tagOrder),
+            "49: " + getCheckMessage(MSG_KEY, tagOrder),
+            "69: " + getCheckMessage(MSG_KEY, tagOrder),
+            "70: " + getCheckMessage(MSG_KEY, tagOrder),
+            "71: " + getCheckMessage(MSG_KEY, tagOrder),
+        };
```

Comment:
New violations found, similar to last grammar change PR, for methods with array return type.